osu.Game/Screens/Select/SongSelect.cs

```diff
@@ -315,8 +319,16 @@ private void load(AudioManager audio, DialogOverlay dialog, OsuColour colours, M
             (new FooterButtonMods { Current = Mods }, ModSelect),
             (new FooterButtonRandom
             {
-                NextRandom = () => Carousel.SelectNextRandom(),
-                PreviousRandom = Carousel.SelectPreviousRandom
+                NextRandom = () =>
+                {
+                    randomClicked = true;
```

Comment:
Really ugly to use a flag like this. I'd move all the sound effects inside the carousel instead, so you don't need to have weird logic like this.

Reply:
Hrm, I'm not sure moving sfx playback to `BeatmapCarousel` would make the logic much cleaner here? Moving playback to `selectNextSet`/`selectNextDifficulty`/`SelectNextRandom` would make the most sense, but then track changes from outside the carousel (i.e. via F5 or the now playing overlay) aren't handled... and implementing in `select` would probably just result in the same funky random flag/playback logic but in a different place?

Reply:
I'd probably add the sample playback to each call site which calls `select()` and call it a day. Would that bring any issues up?

Reply:
Would cause track changes from outside the carousel (i.e. via F5 or the now playing overlay) to no be longer be handled, meaning no audio playback when changing tracks that way.

Reply:
Not sure I follow. Couldn't you add a sample at the following point to handle that case?

https://github.com/ppy/osu/blob/449e9bcf5c123e1e1c92534f8f3a710412a5028d/osu.Game/Screens/Select/BeatmapCarousel.cs#L351

the carousel does see and handle all selection events in one path or another.

Reply:
Problem is that code also gets hit via all the other paths, meaning double playback... To avoid that, I'd pretty much have to do what's already happening in `SongSelect` to determine which sample to play?

Reply:
I'm not seeing the double playback path, but might be missing it. Can you give one example of where this would be the case? ie. a method that you'd want to play a sound on which calls another method that will unavoidably add a second sound?

In case the above isn't clear, I'm not proposing putting the same playback *inside* `select()` but *next* to each usage of it.

ie. adding one `sample.Play()` line at each of the following locations:

https://github.com/ppy/osu/blob/8917ab78f427c1dfb2e5eef47738fafec8fa746a/osu.Game/Screens/Select/BeatmapCarousel.cs#L379-L382

https://github.com/ppy/osu/blob/8917ab78f427c1dfb2e5eef47738fafec8fa746a/osu.Game/Screens/Select/BeatmapCarousel.cs#L467

https://github.com/ppy/osu/blob/8917ab78f427c1dfb2e5eef47738fafec8fa746a/osu.Game/Screens/Select/BeatmapCarousel.cs#L453

https://github.com/ppy/osu/blob/8917ab78f427c1dfb2e5eef47738fafec8fa746a/osu.Game/Screens/Select/BeatmapCarousel.cs#L351

https://github.com/ppy/osu/blob/8917ab78f427c1dfb2e5eef47738fafec8fa746a/osu.Game/Screens/Select/BeatmapCarousel.cs#L391-L394

(potentially a couple of the others too, i'm not focusing on which have unique sounds or sounds at all)

Reply:
Yeah that's essentially what I'm doing, but because of this in `SongSelect`:
https://github.com/ppy/osu/blob/f59828e2d9dc745e352ecbbf01fe811389e2ed47/osu.Game/Screens/Select/SongSelect.cs#L431

Adding playback to handle changes outside `BeatmapCarousel` here:
https://github.com/ppy/osu/blob/8917ab78f427c1dfb2e5eef47738fafec8fa746a/osu.Game/Screens/Select/BeatmapCarousel.cs#L351
will result in a second sample being played (after it has already played in `SelectNext` or wherever triggering `workingBeatmapChanged()`)

Reply:
That seems really weird. I'll take a look, we really shouldn't be calling `select()` on the same beatmap twice, of if we are, it should return a `bool` to allow handling that case.

Reply:
I think I can make this work, but one thing to note is that it would delay the sound along with the selection debounce, which seems to be something you've actively avoided doing. I guess we want to keep the sound activating immediately rather than on debounce (when the audio track changes)?

Reply:
Yeah... IMO it feels much better to have feedback immediately rather than on debounce.

Maybe debounced will be okay for the random sound effect though? If the end goal is to have a roulette-wheel effect?

Reply:
My main goal was to consolidate the location we implement this, so not sure that helps much. I'll figure something out.

Reply:
I'm going to leave things how you have it as I'm not sure on how this will change going forward. As we discussed, an animation is required, and when that happens quite a bit of restructuring is probably going to be required.

I've renamed the flag in line with what we usually name these kinds of things.